exotic/exotic.py

```diff
@@ -1108,57 +1143,38 @@ def exp_time_med(exptimes):
         return np.median(exptimes)
 
 
-# finds target in WCS image after applying proper motion correction from SIMBAD
-def find_target(target, hdufile, verbose=False):
-    # query simbad to get proper motions
-    service = vo.dal.TAPService("http://simbad.u-strasbg.fr/simbad/sim-tap")
-    # http://simbad.u-strasbg.fr/simbad/tap/tapsearch.html
-    query = '''
-    SELECT basic.OID, ra, dec, main_id, pmra, pmdec
-    FROM basic JOIN ident ON oidref = oid
-    WHERE id = '{}';
-    '''.format(target)
-
-    result = service.search(query)
-    # TODO check that simbad returned a value
-
-    # set up astropy object
-    coord = SkyCoord(
-        ra=result['ra'][0] * u.deg,
-        dec=result['dec'][0] * u.deg,
-        distance=1 * u.pc,
-        pm_ra_cosdec=result['pmra'][0] * u.mas / u.yr,
-        pm_dec=result['pmdec'][0] * u.mas / u.yr,
-        frame="icrs",
-        obstime=Time("2000-1-1T00:00:00")
-    )
-
-    hdu = fits.open(hdufile)[0]
-
-    try:
-        dateobs = hdu.header['DATE_OBS']
-    except:
-        dateobs = hdu.header['DATE']
-
-    # ignore timezone
-    if len(dateobs.split('-')) == 4:
-        dateobs = '-'.join(dateobs.split('-')[:-1])
+def update_coordinates_with_proper_motion(info_dict, time_obs):
+    parameter_names = {
+        'dist': 'Distance (pc)',
+        'pm_ra': 'Proper Motion RA (mas/yr)',
+        'pm_dec': 'Proper Motion DEC (mas/yr)'
+    }
 
-    t = Time(dateobs, format='isot', scale='utc')
-    coordpm = coord.apply_space_motion(new_obstime=t)
+    missing_values = [parameter_names[key] for key in ['dist', 'pm_ra', 'pm_dec'] if info_dict.get(key) == 0.0]
```

Comment:
What if `info_dict.get(key)` returns `None` because the key does not exist? Then that's false and the values a ren't replaced. I would suggest `info_dict.get(key. 0.0)`. ... That is unless there's a desire to potentially not have missing keys recognized.

Reply:
Makes sense, I left no default value since I ensure just before this step that it would contain the key. Also, we go through most keys in the same dict and assign values of 0.0 if they are `None`.

Reply:
OK, fair enough.